src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreBuildHandler.cs

```diff
@@ -53,27 +50,32 @@ public sealed class SolutionRestoreBuildHandler
         /// </summary>
         private uint _updateSolutionEventsCookieEx;
 
+        private Microsoft.VisualStudio.Shell.IAsyncServiceProvider _serviceProvider;
+
+        private static SolutionRestoreBuildHandler _instance;
```

Comment:
It's inherently Singleton since package gets loaded once which calls InitializeAsync() It doesn't need to return anything but since that was the existing code so i didn't change this signature.  

There are no cases where this method will be called multiple times.